accepted/2022/simplify-output-paths.md

```diff
@@ -0,0 +1,53 @@
+- `AppendRuntimeIdentifierToOutputPath` will default to true only when targeting .NET 7 or lower.
+  - To be discussed: Should we also default it to true if `RuntimeIdentifiers` is set?
+
+For publish, we would change the publish output path only when targeting .NET 8 or higher and when the Configuration is set to Release (which we [plan to make the default](https://github.com/dotnet/sdk/issues/10357#issuecomment-1208564539) for .NET 8).  In that case we would set the publish output path to `$(BaseOutputPath)\publish` (by default `bin\publish`), and append the `TargetFramework` and `RuntimeIdentifier` to the path based on `AppendTargetFrameworkToOutputPath` and `AppendRuntimeIdentifierToOutputPath`.
+
+## Considerations
+
+### Breaking changes
+
+These changes could break things such as scripts that copy the output of the build or custom MSBuild logic that hard-codes these paths.  Tieing these changes to the project TargetFramework ensures that these breaks will be encountered when the project is modified to target a new TargetFramework, not when updating to a new version of the .NET SDK.
+
+### Why have different output paths at all?
+
+It is worth considering why we have different output paths at all.  The advantage of having different output paths for some pivot (such as configuration, target framework, or RuntimeIdentifier) is that you can keep outputs for the different values of the pivot side-by-side, and that if you are switching back and forth between building them, the builds for a given pivot value can be incremental.
+
+Therefore, the assumption with these changes is that it is not common to switch back and forth between different target frameworks or different runtime identifiers.  Projects where it is common to switch back and forth could still set `AppendTargetFrameworkToOutputPath` or `AppendRuntimeIdentifierToOutputPath` to true.
+
+### Incremental publish
+
+Originally, the .NET SDK did not support incremental publish.  This meant that if you published for one RuntimeIdentifier, and then published for another RuntimeIdentifier using the same publish path, the files from the first RuntimeIdentifier that should not have been included in the second publish would still be left in the folder.  Probably because of this, the RuntimeIdentifier is [always added to the publish output path](https://github.com/dotnet/sdk/blob/efef23ab729388ffb081731e5b1adbabc6e6b327/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets#L122-L126) regardless of the value for `AppendRuntimeIdentifierToOutputPath`.  However, we now [support incremental publish](https://github.com/dotnet/sdk/pull/3957), so this isn't a concern anymore.
+
+### Capitalization
+
+Should the default publish output path be `bin\publish` or `bin\Publish`?  Uppercase matches the Configuration values of `Debug` and `Release`, which the publish folder will be a sibling of.  However, the current publish folder capitalization is lowercase.
```

Comment:
Keep it lowercase, as it's different from a build-config (Debug, Release) so to indicate that difference here too.

Reply:
We have been discussing making Publish be a build-config instead of what it currently is. But I like the lowercase `p` since it matches the rest of the folder conventions.